src/gc/gc.d

```diff
@@ -2911,43 +2915,37 @@ struct LargeObjectPool
      */
     size_t allocPages(size_t n) nothrow
     {
-        if(freepages < n) return OPFAIL;
-        size_t i;
-        size_t n2;
+        if(largestFree < n)
+            return OPFAIL;
 
         //debug(PRINTF) printf("Pool::allocPages(n = %d)\n", n);
-        n2 = n;
-        for (i = searchStart; i < npages; i++)
+        size_t largest = 0;
+        while (searchStart < npages && pagetable[searchStart] != B_FREE)
+            searchStart++;
```

Comment:
Can't you do the `== B_PAGE; searchStart += bPageOffsets[searchStart];` trick here too?